src/Framework/NativeMethods.cs

```diff
@@ -1035,6 +1042,94 @@ internal static MemoryStatus GetMemoryStatus()
         return null;
     }
 
+    internal static bool IsSymLink(string fullPath)
+    {
+#if NET
+        var fileInfo = new FileInfo(fullPath);
+
+        return fileInfo.Exists && fileInfo.Length == 0 && !string.IsNullOrEmpty(fileInfo.LinkTarget);
+#else
+        if (!IsWindows)
+        {
+            return false;
+        }
+
+        WIN32_FILE_ATTRIBUTE_DATA data = new WIN32_FILE_ATTRIBUTE_DATA();
+
+        return NativeMethods.GetFileAttributesEx(fullPath, 0, ref data) &&
+               (data.fileAttributes & NativeMethods.FILE_ATTRIBUTE_DIRECTORY) == 0 &&
+               (data.fileAttributes & FILE_ATTRIBUTE_REPARSE_POINT) == FILE_ATTRIBUTE_REPARSE_POINT;
+#endif
+    }
+
+    internal static void ReadFileThroughSymlinks(string filePath, Action<FileStream> contentConsumer)
+    {
+        filePath = Path.GetFullPath(filePath);
+
+        if (IsSymLink(filePath))
+        {
+            using SafeFileHandle handle = OpenFileThroughSymlinks(filePath);
```

Comment:
For these sorts of errors, you should be able to add `[SupportedOSPlatform("windows")]` to the relevant methods.